lib/improver/nowcasting/lightning.py

```diff
@@ -0,0 +1,455 @@
+                                                                 this_time)
+            assert isinstance(high_precip,
+                              iris.cube.Cube), err_string.format("high precip",
+                                                                 this_time)
+            assert isinstance(torr_precip,
+                              iris.cube.Cube), err_string.format(
+                                  "intense precip",
+                                  this_time)
+            # Increase prob(lightning) to Risk 2 (pl_dict[2]) when
+            #   prob(precip > 7mm/hr) > phighthresh
+            cube_slice.data = np.where(
+                (high_precip.data >= self.phighthresh) &
+                    (cube_slice.data < self.pl_dict[2]),
+                self.pl_dict[2], cube_slice.data)
+            # Increase prob(lightning) to Risk 1 (pl_dict[1]) when
+            #   prob(precip > 35mm/hr) > ptorrthresh
+            cube_slice.data = np.where(
+                (torr_precip.data >= self.ptorrthresh) &
+                    (cube_slice.data < self.pl_dict[1]),
+                self.pl_dict[1], cube_slice.data)
+
+            # Decrease prob(lightning) where prob(precip > 0) is low.
+            cube_slice.data = apply_double_scaling(
+                this_precip, cube_slice, self.precipthr, self.ltngthr)
+
+            new_cube_list.append(cube_slice)
+
+        new_cube = new_cube_list.merge_cube()
+        new_cube = check_cube_coordinates(
+            prob_lightning_cube, new_cube)
+        return new_cube
+
+    def apply_ice(self, prob_lightning_cube, ice_cube):
+        """
+        Modify Nowcast of lightning probability with ice data from radarnet
+        composite (VII; Vertically Integrated Ice)
+
+        Args:
+            prob_lightning_cube (iris.cube.Cube):
+                First-guess lightning probability.
+                This is modified in-place.
+            ice_cube (iris.cube.Cube):
+                Analysis of vertically integrated ice (VII) from radar
+
+        Returns:
+            new_cube (iris.cube.Cube):
+                Output cube containing updated nowcast lightning probability.
+                This cube will have the same dimensions and meta-data as
+                prob_lightning_cube.
+        """
+        prob_lightning_cube.coord('forecast_period').convert_units('minutes')
+        # check prob-ice threshold units are as expected
+        ice_cube.coord('threshold').convert_units('kg m^-2')
+        new_cube_list = iris.cube.CubeList([])
+        for cube_slice in prob_lightning_cube.slices_over('time'):
+            fcmins = cube_slice.coord('forecast_period').points[0]
+            for threshold, prob_max in zip(self.ice_thresholds,
+                                           self.ice_scaling):
+                ice_slice = ice_cube.extract(
+                    iris.Constraint(threshold=threshold))
+                ice_scaling = [0., (prob_max * (1. - (fcmins / 150.)))]
```

Comment:
Where does this come from?  Curious why this is defined here rather than as a class member, unless it's a property of the VII value (eg unit conversion); in which case a comment would be helpful.